From 1a781819811a1251b77e0f9fc2843e65b432ea7c Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Tue, 2 Jan 2018 08:40:32 +0000
Subject: [PATCH] wr-policy: allow ping_t read all levels' file

Allow ping_t to read files with level s15:c0.c1023/SystemHigh.

Fix avc denial:
  avc: denied { read } for pid=481 comm="ping" \
  name="ld.so.cache" dev="sda1" ino=1063343 \
  scontext=root:sysadm_r:ping_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:etc_t:s15:c0.c1023 tclass=file

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/admin/netutils.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/admin/netutils.te b/policy/modules/admin/netutils.te
index 3bfb406..8db03d6 100644
--- a/policy/modules/admin/netutils.te
+++ b/policy/modules/admin/netutils.te
@@ -131,6 +131,7 @@ domain_use_interactive_fds(ping_t)
 
 files_read_etc_files(ping_t)
 files_dontaudit_search_var(ping_t)
+mls_file_read_all_levels(ping_t)
 
 kernel_read_system_state(ping_t)
 
-- 
2.13.3

